### PR TITLE
DOC: Add examples for integrals of Bessel functions

### PR DESCRIPTION
**Pull request by dschmitz89**
#### Reference issue
Example: Towards #7168 

#### What does this implement/fix?
Adds examples to [integrals](https://docs.scipy.org/doc/scipy/reference/special.html#integrals-of-bessel-functions) of Bessel functions.

[skip azp][skip actions]

### CODE REVIEW

**Comment by WarrenWeckesser** [comment]
Thanks @dschmitz89.

Just a "drive-by comment" for the moment: we generally don't put a blank line between the section heading and the first line of content in the section.  You can see this in the [numpydoc guidelines](https://numpydoc.readthedocs.io/en/latest/format.html#examples), and it looks like you already following the recommended convention in gh-17003.

File: scipy/special/_add_newdocs.py [hunk 1]
```diff
@@ -1142,6 +1142,41 @@ def add_newdoc(name, doc):
...
+    Plot the function for varying powers, orders and scales.
+
+    >>> import matplotlib.pyplot as plt
+    >>> powers = np.linspace(0, 10, 100)
+    >>> orders = [1, 2, 3]
+    >>> scales = [1, 2]
+    >>> all_combinations = [(order, scale) for order in orders
+    ...                     for scale in scales]
+    >>> for order, scale in all_combinations:
+    ...     plt.plot(powers, besselpoly(scale, powers, order),
```
Comment (tupui) [inline scipy/special/_add_newdocs.py:1174]: Can you refactor these to use the OOP version of matplotlib instead? We should follow matplotlib's guidelines here. (I have seen that in other PR's IIRC)

**Review by tupui — CHANGES_REQUESTED** [review]
Thank you for adding these examples @dschmitz89!

**Review by tupui — CHANGES_REQUESTED** [review]
Thanks for the update and for caring about the CI! Adding some nitpick so that we can also run with Azure as we need to check the refguide (actions can still be skipped).

**Comment by dschmitz89** [comment]
> Thanks for the update and for caring about the CI! Adding some nitpick so that we can also run with Azure as we need to check the refguide (actions can still be skipped).

Would it be possible to add sth like `[ci docs]` to the CI machinery which only executes ref-guide, linter and the doc builder? Otherwise a lot of azure CI resources are used for no reason.

**Comment by tupui** [comment]
We should totally have better flags there 👍 The distinction with provider instead of tasks makes little sense. I proposed some reflexion but it did not do anything. If you have some concrete proposal we could share this on the mailing list and I would gladly add my +1 😃

**Review by tupui — APPROVED** [review]
LGTM, thanks again @dschmitz89.

**Pull request merged**